### PR TITLE
feat: add getClientToken

### PR DESCRIPTION
**Pull request by fallion**
- allows for getting the client token which is the last step needed to make requests to IAP protected endpoints

### CODE REVIEW

File: src/getClientToken.test.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { DesktopIAPOptions } from "./types";
+
+test("getClientToken", async (t) => {
+  let nFetchCalls = 0;
+
+  const mockFetch = async (): Promise<Response> => {
+    nFetchCalls += 1;
+    return new Response(
+      JSON.stringify({
+        // eslint-disable-next-line @typescript-eslint/camelcase
```
Comment (nglgzz) [inline src/getClientToken.test.ts]: Do we still need this line? I thought we updated eslint rules to disable this error.

File: src/getClientToken.test.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const mockIAPOptions: DesktopIAPOptions = {
+    clientId: "Test",
+    clientSecret: "TestSecret",
+    iapClientId: "testiapClient",
+  };
+
+  const clientToken = await getClientToken(
+    mockFetch,
+    mockIAPOptions,
+    "testrefrshtoken",
```
Comment (nglgzz) [inline src/getClientToken.test.ts]:
```suggestion
    "testrefreshtoken",
```

File: src/getClientToken.test.ts (hunk 1 above)
Comment (fallion) [reply src/getClientToken.test.ts]: It's still throwing errors.

**Pull request merged**